pkgs/development/python-modules/patool/default.nix

```
@@ -0,0 +1,58 @@
+{ stdenv, lib, buildPythonPackage, fetchFromGitHub, pytest, p7zip, unzip,
+  cabextract, zip, zopfli, lzip, zpaq, gnutar, gnugrep, diffutils, file,
+  gzip, bzip2, xz}:
+
+# unrar is unfree, as well as 7z with unrar support, not including it (patool doesn't support unar)
+
+let
+  compression-utilities = [
+    p7zip
+    unzip
+    gnutar
+    cabextract
+    zip
+    zopfli
+    lzip
+    zpaq
+    gzip
+    gnugrep
+    diffutils
+    bzip2
+    file
+    xz
+  ];
+in
+buildPythonPackage rec {
+  pname = "patool";
+  version = "1.12";
+
+  #pypi doesn't have test data
+  src = fetchFromGitHub {
+    owner = "wummel";
+    repo = pname;
+    rev = "upstream/${version}";
+    sha256 = "0v4r77sm3yzh7y1whfwxmp01cchd82jbhvbg9zsyd2yb944imzjy";
+  };
+
+  prePatch = ''
+    substituteInPlace patoolib/util.py \
+      --replace "path = None" 'path = append_to_path(os.environ["PATH"], "${lib.makeBinPath compression-utilities}")'
+  '';
+
+  checkInputs = [ pytest ] ++ compression-utilities;
+
+  # for some reason, zip can't decompress file that doesn't finish by .zip, but patool doesn't take this in account
+  checkPhase = ''
```

Per https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/explicit-phases.md 

```suggestion
  preCheck = ''
```